src/lang/base/Syntax.ml

```diff
@@ -499,6 +500,62 @@ module ScillaSyntax (SR : Rep) (ER : Rep) = struct
         let body' = subst_type_in_expr tvar tp body in
         (Fixpoint (f, t', body'), rep)
 
+(* Is expr dependent on any ident in blist.
+ * This is the same as checking if a free var
+ * in expr is present in blist. *)
+  let free_vars_dep_check erep blist =
+
+    (* is m in l. *)
+    let is_mem m l =
+      List.exists l ~f:(fun x -> get_id m = get_id x) in
+    (* is any m in ml, in l. *)
+    let any_is_mem ml l =
+      List.exists ml ~f:(fun i -> is_mem i l) in
+    (* get elements in "l" that are not in bound_vars. *)
+    let get_free l bound_vars =
+      List.filter l ~f:(fun i -> not (is_mem i bound_vars)) in
+    let get_pattern_bounds p =
+      let rec accfunc p acc =
+        match p with
+        | Wildcard -> []
+        | Binder i -> i::acc
+        | Constructor (_, plist) ->
+          List.fold plist ~init:[] ~f:(fun acc p' -> accfunc p' acc)
```

Comment:
I think the `Wildcard` case should return `acc`, and in the `Constructor` case `~init` should be `acc`. I'll fix it and merge.